<launch>
  <node pkg="ltme_node" type="ltme_node" name="ltme_node" output="screen">
    <param name="device_model" value="LTME-02A"/>

    <!-- IP address of LTME-02 -->
    <param name="device_address" value="192.168.10.160"/>

    <!-- Enforce specific transport mode used by the device to stream measurement data. -->
    <!-- If set to "normal" or "oob", device settings will be changed to selected mode in case of a mismatch. -->
    <!-- Available options are: -->
    <!-- * none: don't enforce any mode and use device's current configuration -->
    <!-- * normal: enforce normal (in-band) mode, i.e. TCP connection for command interaction is reused for data streaming -->
    <!-- * oob: enforce out-of-band mode, where data are streamed through a dedicated UDP channel -->
    <!-- <param name="enforced_transport_mode" value="none"/> -->

    <!-- Enforce the device to scan at specified frequency. -->
    <!-- Available options are: 0 (disable scan frequency enforcement), 10, 15, 20, 25 and 30. -->
    <!-- If this option is enabled (i.e., set to a non-zero value), and the device is running at a different frequency, -->
    <!-- the enforced value will be set (and saved in non-volatile memory) as new frequency; -->
    <!-- then the device will be rebooted for once for the new settings to take effect. -->
    <!-- <param name="enforced_scan_frequency" value="0"/> -->

    <!-- Frame ID used by the published LaserScan messages -->
    <!-- <param name="frame_id" value="laser"/> -->

    <!-- If this option is enabled, published LaserScan messages will have their X and Z axes inverted. -->
    <!-- This is mostly useful when the device is mounted upside down, as it effectively undos the inversion created by the mounting, -->
    <!-- and makes it look like the scans are from a device installed in a normal, upward direction. -->
    <!-- <param name="invert_frame" value="false"/> -->

    <!-- Start and end angle of published scans -->
    <!-- As LTME-02 has an FOV of 270 degrees, the minimum allowed value for angle_min is -2.356 (about -3 * pi / 4), and the maximum allowed value for angle_max is 2.356 (about 3 * pi / 4) -->
    <!-- <param name="angle_min" value="-1.571"/> -->
    <!-- <param name="angle_max" value="1.571"/> -->

    <!-- Range of angle for which data should be excluded from published scans -->
    <!-- Leave these two parameters commented out if a full 270 degree FOV is desired -->
    <!-- <param name="angle_excluded_min" value="-0.785"/> -->
    <!-- <param name="angle_excluded_max" value="0.785"/> -->

    <!-- Minimum and maximum range value of published scans -->
    <!-- Defaults to 0.05 and 30 respectively if not specified -->
    <!-- <param name="range_min" value="0.05"/> -->
    <!-- <param name="range_max" value="30"/> -->

    <!-- Number of neighboring measurements to be averaged -->
    <!-- Averaging reduces jitter but angular resolution will also decrease by the same factor -->
    <!-- <param name="average_factor" value="2"/> -->

    <!-- Threshold value of tolerated background light intensity -->
    <!-- Valid values are between 0 and 4095 inclusive -->
    <!-- <param name="background_intensity_threshold" value="256"/> -->
  </node>
</launch>
